wrappers/csharp/Managed.cs

```diff
@@ -862,6 +861,55 @@ public static string DecryptWithPasswordAsString(string b64data, string password
             return DecryptWithPasswordAsUtf8String(b64data, password, iterations);
         }
 
+        /// <summary>
+        /// Join multiple shares to regenerate a shared secret.
+        /// </summary>
+        /// <param name="shares">This is a 2-dimensional array representing the shares.</param>
+        /// <returns>The output buffer with the shared secret.</returns>
+        public static byte[] JoinShares(byte[][] shares)
+        {
+            if (shares == null || shares.Length == 0 || !SharesLengthAreValid(shares))
```

Comment:
shares == null and shares.Length == 0 is validated 2 times, so you can remove the checks here and leave only !SharesLengthAreValid(shares)